restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java

```diff
@@ -121,6 +139,11 @@ private JsonNode convertToJsonDoc(
         if (isArray && !writeAllPathsAsObjects) {
           int index = Integer.parseInt(p.substring(1, p.length() - 1));
 
+          boolean shouldBeArray = isArray && !ref.isArray();
+          if (i == 0 && shouldBeArray) {
```

Comment:
why onlt if `i == 0`?

Reply:
Good point, seems that we would end up in the cast exception if `i` is not `0` and it should be array?

Reply:
If entries are sorted first, wouldn't we always get `[0]` entry first? Unless that was filtered I guess?
Then again there should be no harm in checking it for further entries either way.

Reply:
So I took a look at this code, and yes - for completeness there should be a clause to deal with converting an object to an array when it is not at the root - I added that.

However, because of how row deletions occur before inserts in the API, there isn't a real situation at the moment that causes an object to have to be converted to an array - which is why this issue has never been hit. At any rate, I added a test that should cover this behavior, as well, in case something changes in the future to how we delete data before writing.

Reply:
ahhh. My comment was in error; I did not pay attention and was thinking of 0 as the first element, as opposed to the first path segment.